### DISCUSSION TITLE
Wanted: ability to remediate vulnerable indirect go dependencies without updating all indirect

### DISCUSSION THREAD

**Discussion by cevich** [renovatebot/renovate#22599] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'd prefer Renovate to propose potentially broken golang dep. update PRs for indirect go module vulnerabilities, rather than skip them by default.  Based on [recent experience](https://github.com/containers/podman/pull/18709), Dependabot **does** propose these updates (when configured).  I believe renovate skips them because [the default golang setting is to ignore indirect dependencies](https://docs.renovatebot.com/modules/manager/gomod/#post-update-options) (normally, probably a good thing).

Note: This is likely at least loosely related to [the "nested" `go mod tidy` request](https://github.com/renovatebot/renovate/issues/12999).  Based on my (limited) understanding, a fix for that would be needed for golang vulnerability update PRs due to indirect deps.

### What you tried so far.

I *think* this might work, but am not sure:

```json5
  "vulnerabilityAlerts": {
    "enabled": true
    "packageRules": [
      {
        "matchManagers": ["gomod"],
        "matchDepTypes": ["indirect"],
        "enabled": true,
      }
    ]
  },
```

### Relevant debug logs

<details><summary>Alert</summary>

```
{
  "alerts": {
    "go/github.com/sigstore/rekor": {
      "< 1.2.0": "1.2.0"
    }
  }
}
```

</details>

<details><summary>Details</summary>

```
{
  "alertPackageRules": [
    {
      "matchDatasources": [
        "go"
      ],
      "matchPackageNames": [
        "github.com/sigstore/rekor"
      ],
      "matchCurrentVersion": "= 1.1.2-0.20230508234306-ad288b385a44",
      "matchFiles": [
        "go.mod"
      ],
      "allowedVersions": "1.2.0",
      "prBodyNotes": [
        "### GitHub Vulnerability Alerts",
        "#### [CVE-2023-33199](https://github.com/sigstore/rekor/security/advisories/GHSA-frqx-jfcm-6jjr)\n\n### Impact\nA malformed proposed entry of the `intoto/v0.0.2` type can cause a panic on a thread within the Rekor process. The thread is recovered so the client receives a 500 error message and service still continues, so the availability impact of this is minimal.\n\n### Patches\nThis is fixed in v1.2.0 of Rekor.\n\n### Workarounds\nNo\n\n### References\nDiscovered by OSS-Fuzz"
      ],
      "isVulnerabilityAlert": true,
      "force": {
        "groupName": null,
        "schedule": [],
        "dependencyDashboardApproval": false,
        "minimumReleaseAge": null,
        "rangeStrategy": "update-lockfile",
        "commitMessageSuffix": "[SECURITY]",
        "branchTopic": "{{{datasource}}}-{{{depName}}}-vulnerability",
        "prCreation": "immediate",
        "labels": [
          "dependencies",
          "security"
        ]
      }
    }
  ]
}
```

</details>

<details><summary>Indirect dep. disabled</summary>

```
          {
            "currentDigest": "ad288b385a44",
            "currentValue": "v1.1.2-0.20230508234306-ad288b385a44",
            "datasource": "go",
            "depName": "github.com/sigstore/rekor",
            "depType": "indirect",
            "digestOneAndOnly": true,
            "enabled": false,
            "managerData": {
              "lineNumber": 153,
              "multiLine": true
            },
            "packageName": "github.com/sigstore/rekor",
            "skipReason": "disabled",
            "updates": []
          },
```

</details>

**Comment by rarkins** [comment]
Does the new title I made reflect what you're trying to ask for?

**Reply by cevich** [reply]
Yes, perfect.  Thanks.

**Comment by Churro** [comment, marked as answer]
Could you please add the following snippet to your config? It will _force-enable_ disabled deps in case of vulnerability alerts:
```json
"vulnerabilityAlerts": {
  "enabled": true
}
```

**Reply by cevich** [reply]
Oh!  That's a really good suggestion.  I never would have thought to do that.  Okay, done.

**Reply by cevich** [reply]
Though actually...there are some cases where we _really_ do want to disable renovate management - for example, `Dockerfiles` that are **only** used for build-testing and should never ever be touched -  since they're strictly tied to very specific build-test cases and no image is ever actually pushed/published.

Would this `enabled: true` setting override any `ignorePaths` items that happen to be set?

**Reply by rarkins** [reply]
It wouldn't/couldn't change which files are *extracted* in the first place

**Reply by rarkins** [reply]
It wouldn't/couldn't change which files are *extracted* in the first place

**Reply by Churro** [reply]
No, it wouldn't affect paths listed in `ignorePaths`. Deps in those paths aren't even extracted in the first place. 

There won't be any security PRs for Docker images. For regular deps that you want to explicitly exempt from vuln fix PRs, you can still specify package rules within `vulnerabilityAlerts` that would disable the force-enabling again based on the principle: _last rule wins_.

**Reply by cevich** [reply]
Great, thanks @Churro and @rarkins I appreciate it.
